src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -911,4 +1095,108 @@ public sealed override int GetHashCode()
+            ushort maxStack,
+            ImmutableArray<byte> il)
+            : base(containingType, maxStack, il)
+        {
+            _parameters =
+            [
+                new BytesParameter(encodingGetString),                              // byte* bytes
+                new ParameterDefinition(1, "length", Cci.PlatformType.SystemInt32), // int length
+            ];
+        }
+
+        public static BytesToStringHelper Create(
+            ITokenDeferral moduleBuilder,
+            Cci.INamespaceTypeDefinition containingType,
+            Cci.IMethodReference encodingUtf8,
+            Cci.IMethodReference encodingGetString,
+            DiagnosticBag diagnostics)
+        {
+            var ilBuilder = new ILBuilder(
+                moduleBuilder,
+                new LocalSlotManager(slotAllocator: null),
+                OptimizationLevel.Release,
+                areLocalsZeroed: false);
+
+            // Call `Encoding.get_UTF8()`.
+            ilBuilder.EmitOpCode(ILOpCode.Call, 1);
+            ilBuilder.EmitToken(encodingUtf8, null, diagnostics);
+
+            // Push the `byte*`.
+            ilBuilder.EmitOpCode(ILOpCode.Ldarg_0);
+
+            // Push the byte size.
+            ilBuilder.EmitOpCode(ILOpCode.Ldarg_1);
+
+            // Call `Encoding.GetString(byte*, int)`.
+            ilBuilder.EmitOpCode(ILOpCode.Callvirt, -2);
+            ilBuilder.EmitToken(encodingGetString, null, diagnostics);
+
+            // Return.
+            ilBuilder.EmitRet(isVoid: false);
+            ilBuilder.Realize();
+
+            return new BytesToStringHelper(
+                containingType: containingType,
+                encodingGetString: encodingGetString,
+                maxStack: ilBuilder.MaxStack,
+                il: ilBuilder.RealizedIL);
+        }
+
+        public override string Name => PrivateImplementationDetails.SynthesizedBytesToStringFunctionName;
+        public override Cci.TypeMemberVisibility Visibility => Cci.TypeMemberVisibility.Private;
+        public override ImmutableArray<Cci.IParameterDefinition> Parameters => _parameters;
+        public override Cci.ITypeReference GetType(EmitContext context) => context.Module.GetPlatformType(Cci.PlatformType.SystemString, context);
+
+        private sealed class BytesParameter(
+            Cci.IMethodReference encodingGetString)
+            : Cci.ParameterDefinitionBase
+        {
+            private readonly Cci.IMethodReference _encodingGetString = encodingGetString;
+
+            public override ushort Index => 0;
+            public override string Name => "bytes";
+            public override Cci.ITypeReference GetType(EmitContext context)
+            {
+                return _encodingGetString.GetParameters(context)[0].GetType(context);
+            }
+        }
+    }
+
+    file sealed class ParameterDefinition(
+        ushort index,
+        string name,
+        Cci.PlatformType type)
```

Comment:
>Cci.PlatformType type [](http://example.com/codeflow?start=8&length=21)

It looks like we always pass ```Cci.PlatformType.SystemInt32```, similar with name, it looks like it is always the same. Let's get rid of the parameters if that is the case

Reply:
```index``` as well?